disperser/dataapi/metrics.go

```diff
@@ -101,35 +104,124 @@ type DynamoDBCollector struct {
 	blobMetadataStore *blobstore.BlobMetadataStore
 	blobStatusMetric  *prometheus.Desc
 	logger            common.Logger
+	metricsCache      map[disperser.BlobStatus]int
+	cacheMutex        sync.RWMutex
 }
 
 func NewDynamoDBCollector(blobMetadataStore *blobstore.BlobMetadataStore, logger common.Logger) *DynamoDBCollector {
-	return &DynamoDBCollector{
+	collector := &DynamoDBCollector{
 		blobMetadataStore: blobMetadataStore,
 		blobStatusMetric: prometheus.NewDesc("dynamodb_blob_metadata_status_count",
 			"Number of blobs with specific status in DynamoDB",
 			[]string{"status"},
 			nil,
 		),
-		logger: logger,
+		logger:       logger,
+		metricsCache: make(map[disperser.BlobStatus]int),
 	}
+
+	if blobMetadataStore != nil {
+		go collector.periodicFetch()
+	} else {
+		logger.Warn("BlobMetadataStore is nil, metrics will not be collected")
+	}
+
+	return collector
 }
 
-func (collector *DynamoDBCollector) Describe(ch chan<- *prometheus.Desc) {
-	ch <- collector.blobStatusMetric
+func (collector *DynamoDBCollector) periodicFetch() {
+	// We don't want to fetch finalized status because the amount is large
+	statuses := []disperser.BlobStatus{disperser.Processing, disperser.Confirmed, disperser.InsufficientSignatures, disperser.Failed}
+
+	startFetch := func(status disperser.BlobStatus) {
+		go func() {
+			collector.updateMetricsCache(status)
+		}()
+	}
+
+	// Start the first fetch for each status immediately.
+	for _, status := range statuses {
+		collector.logger.Info("Fetching blob metadata by status", "status", status)
+		startFetch(status)
+	}
+
+	// Then start the ticker for subsequent fetches.
+	ticker := time.NewTicker(time.Minute)
+	defer ticker.Stop()
+
+	for range ticker.C {
+		for _, status := range statuses {
+			collector.logger.Info("Fetching blob metadata by status", "status", status)
+			startFetch(status)
+		}
+	}
 }
 
-func (collector *DynamoDBCollector) Collect(ch chan<- prometheus.Metric) {
-	count, err := collector.blobMetadataStore.GetBlobMetadataByStatusCount(context.Background(), disperser.Processing)
+// updateMetricsCache calls getBlobMetadataByStatus to fetch the count of blob
+// metadata by status, and then updates the metricsCache with the new counts.
+func (collector *DynamoDBCollector) updateMetricsCache(status disperser.BlobStatus) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
```

Comment:
use a context from parent instead of creating a new context each time with `context.Background()`?